### DISCUSSION TITLE
Airflow 2.3: Timeline?

### DISCUSSION THREAD

**Discussion by p-hebert** [apache/airflow#21172] [Q&A]
Hey there! 👋 

Our team has a close interest in the release of Airflow 2.3, as we are currently experiencing the issue described in (#19699 (PR=#19769)) with 2.2.3, which results in requiring human intervention to ensure that our daily deliverables reach our clients on time (6AM EST).

Luckily we've been having the help of a wonderful team member in CET to help us ensure that our client keep receiving our deliverables on time without having us to wake up at unholy morning hours.

Unfortunately, the person responsible for watching & fixing the issue described in #19699 manually daily will be leaving our team soon, as of Feb 28th. We are a small team and we do not have the resources at the moment to hire an extra person in CET tz to carry out those duties.
We are a bit anxious that the 2.3 release does not make it on time, and that we end up being in a difficult situation.
We already have a contingency plan in place (see #21174), but we'd still prefer having the 2.3 release in as well.

Hence the question of this post:
Is it possible for the Apache airflow team to ensure that the milestone of Feb 14th is respected, even if that means splitting 2.3 in two releases (2.3 + 2.3.1 or 2.3 + 2.4)?

Hopefully this does not sound too egregious, but you know, "you miss 100% of the shots you don't take" 😅 

Thanks a lot,
Cheers,
Philippe

---
Disclaimer: The author of #19699 is part of our team.

**Comment by potiuk** [comment]
The Feb 14 is really indicative - it's not a promise. My personal view is that we "release when we are ready" - there are a number of features in-progres for 2.3.0 and we usually do (with the help of Astronomer team mostly) perform a lot of manual regression testing when we release a minor release. This minor relase introduces quite a few things so it is unlikely we speed it up.

And I do not think single user problem will make us change the schedule. Splitting it to two would mean also breaking the SemVer promise that we made to our users (new features only on in MINOR release). So 2.3 and 2.3.1 are totally out of question. Also splitting to to 2.3 and 2.4 is not a good idea as there is significant effort involved in MINOR release testing - much more than patchlevel). And it's more af a joint "team" decision whether we are ready to realease. 

But when you look at the problem/PR, I think the change is isolated and small enough (and also the bug is easily reproducible so it can be easily tested), and has a very good unit test coverage, so I would say it might be a candidate to cherry-pick that one to the upcoming 2.2.4 (which is scheduled to happen early in February). 

But this is just my POV I think @jedcunningham who is the release manager and @ephraimbuddy  who made the fix and tested the change are better to make the call.

**Reply by ephraimbuddy** [reply]
Agree. @philippefutureboy can you cherry-pick the commit and test it in your deployment? That will be very much appreciated. Thanks

**Comment by p-hebert** [comment]
Hey @potiuk, @ephraimbuddy !

Thanks for your thorough response. I understand your circumstances and process and I understand it is not something that can be moved/changed easily.
At least I asked :)

I'll open an issue in our internal task tracker to cherry pick the commit to our deployment of airflow and see how that tests out.
Just to be clear, this implies:

1. Cloning repository locally & testing cherry-picking against the tag we are currently using in our setup
2. Copy the Dockerfile for the tag we are currently using in our setup (`apache/airflow:2.2.3-python3.8` I believe) and perform the same modifications against the repo in the image
3. Test this in our deployment.

Did I miss anything?

**Reply by jedcunningham** [reply]
Another option is to just grab the patch from that PR (e.g. https://github.com/apache/airflow/pull/19769.patch) and patch the image instead. Less moving parts that way 🤷‍♂️

(Also, knowing how to do this sort of thing so you can fix issues for your particular environment faster than the community can is _always_ good)

**Reply by p-hebert** [reply]
Any tutorial/guide on how I can patch an image like you say?
I don't know how to perform the operation you speak of

**Reply by jedcunningham** [reply]
More or less, something like this in your Dockerfile after you've removed the test changes from the patch file:

```
USER root
RUN apt update && apt install -y patch && apt clean
USER 50000
COPY some.patch /tmp
RUN cd /home/airflow/.local/lib/python3.7/site-packages/airflow && patch -u -p2 < /tmp/some.patch
```

**Reply by p-hebert** [reply]
That sounds like some fantastic wizardry! I'll check it out :)

**Reply by potiuk** [reply]
Oh yeah. Learning this kind of wizardry is highly appreciated among our users as they have great opportunity to help us with verifying that what we THINK fix the problems, actually DOES :).

Let us know if that works for you :)

**Reply by p-hebert** [reply]
@jedcunningham @potiuk @ephraimbuddy Our initial smoketests have been positive. We'll deploy in staging/production in the next few days. I'll keep you up to date
